From: Your Name <you@example.com>
Date: Sun, 26 Feb 2023 19:46:04 +0000
Subject: Disable csp reports

Prevents submitting CSP Reports to websites. More info here: https://github.com/gorhill/uBlock/wiki/Dashboard:-Settings#block-csp-reports.

Initial Commit: https://github.com/uazo/cromite/commit/beade82b64cc6ded6a7efd7722bd86a3a4faffc0#diff-fe63966be511427cfd2ab15fe1663e8c02f5a15c4d91fdd1a86ef19f3d4015d2
Category: Privacy
---
 third_party/blink/renderer/core/loader/ping_loader.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/third_party/blink/renderer/core/loader/ping_loader.cc b/third_party/blink/renderer/core/loader/ping_loader.cc
--- a/third_party/blink/renderer/core/loader/ping_loader.cc
+++ b/third_party/blink/renderer/core/loader/ping_loader.cc
@@ -68,6 +68,7 @@ bool SendBeaconCommon(const ScriptState& state,
                       LocalFrame* frame,
                       const KURL& url,
                       const BeaconData& beacon) {
+  if ((true)) return true;
   if (!frame->DomWindow()
            ->GetContentSecurityPolicyForWorld(&state.World())
            ->AllowConnectToSource(url, url, RedirectStatus::kNoRedirect)) {
@@ -107,6 +108,7 @@ bool SendBeaconCommon(const ScriptState& state,
 void PingLoader::SendLinkAuditPing(LocalFrame* frame,
                                    const KURL& ping_url,
                                    const KURL& destination_url) {
+  if ((true)) return;
   if (!ping_url.ProtocolIsInHTTPFamily())
     return;
 
@@ -147,6 +149,7 @@ void PingLoader::SendLinkAuditPing(LocalFrame* frame,
 void PingLoader::SendViolationReport(ExecutionContext* execution_context,
                                      const KURL& report_url,
                                      scoped_refptr<EncodedFormData> report) {
+  if ((true)) return;
   ResourceRequest request(report_url);
   request.SetHttpMethod(http_names::kPOST);
   request.SetHTTPContentType(AtomicString("application/csp-report"));
--
